### ISSUE TITLE
Add support for yarn v2+ 

### ISSUE THREAD

**Issue by jagamypriera** [ds300/patch-package#496]
While yarn v2 has a built-in patch feature, but it involves a lengthy process. To be honest, patch-package is much easier and more convenient to use. Please consider supporting Yarn v2+.

**Comment by JounQin** [comment]
+1 for this, the patch experience of yarn 2+ is bad as pnpm, I love `patch-package`'s solution.

**Comment by noahm** [comment]
I agree, `patch-package` has a much better UX and I prefer using it over yarn's native patches by far. To that end, I already have mostly working support for yarn 2+ in a fork and I hope to eventually get it into shape for a PR here.

**Comment by JounQin** [comment]
> I already have mostly working support for yarn 2+ in a fork and I hope to eventually get it into shape for a PR here.

@noahm Would you like to share it? Or just raise it as a PR?

**Comment by noahm** [comment]
@JounQin Thanks for reminding me! I just opened #506 with my changes. I wouldn't personally consider it ready to merge until new integration tests were added with more recent yarn versions, but I don't have time to keep working on it for a while. If anybody else wants to help along those lines, it would be appreciated!

**Comment by JounQin** [comment]
I raised a new PR #507 to fix all my known issues.

**Comment by JounQin** [comment]
I've just publish my PR #507 as `@unts/patch-package`, you can give it a try.
